### PR TITLE
Jetpack focus jetpack powered bottom sheet

### LINKED ISSUE #16974: [Android] Add "Jetpack powered" overlay

### PR DESCRIPTION
**Pull request by ravishanker**
This PR introduces **Jetpack powered** bottom sheet to be invoked from Jetpack badges, banners and tabs.  

Fixes: #16974 & #16978

**NOTE**:  Consider this as part 1 to keep the PR manageable.  It is only hooked to Jetpack badge on home tab for testing.  Although a full screen version is in implementation, it is not complete and not hooked to Reader, Stats and Notifications tabs yet.  It'll be updated in the next iteration to proper layout and styles subject to final discussions.

https://user-images.githubusercontent.com/990349/179146377-61ffb43d-d252-4754-99ed-172a593f6fc2.mp4

To test:

Setup:

- Launch WordPress app
- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find JetpackPoweredFeatureConfig under Features in development as shown on this PR #16908
- Re-launch app (scroll down if necessary)

Test 1:

- Make sure the Jetpack powered badge appears on the home tab as shown on this PR #16901 
- Tap on the Jetpack powered badge, Ensure a bottom sheet appears as shown in video above
- Observe Logo animation 

Test 2:

- Tap on Get the Jetpack app button, it should open Play Store for Jetpack app

Test 3:

- Return to home, tap on badge again.  When bottom sheet opens 
- Try dragging down and ensure it closes.
- Return to home, tap on badge again.  When bottom sheet opens, tap anywhere outside, ensure bottom sheet closes.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ovitrif — COMMENTED** [review]
It works great and the code looks great to me 🎉 🥇 💯 !

Awesome work @ravishanker!

I have 2 remarks:
1. The night mode coloring is not (yet) implemented, would it be ok to adapt the colors in this PR?

|  |
| --- |
2. Should we add a feature flag for making the Jetpack Powered badges & banners actionable in an upcoming PR, so that we can release the non-clickable work without any blockers?
I'm a bit afraid of the risk of not having enough time to make all badges & banners actionable in time for next release.

**Comment by ravishanker** [comment]
> It works great and the code looks great to me 🎉 🥇 💯 !
> 
> Awesome work @ravishanker!
> 
> I have 2 remarks:
> 
>     1. The night mode coloring is not (yet) implemented, would it be ok to adapt the colors in this PR?

 I need to extract styles, so will handle that in the next PR

>     2. Should we add a feature flag for making the Jetpack Powered badges & banners actionable in an upcoming PR, so that we can release the non-clickable work without any blockers?
>        I'm a bit afraid of the risk of not having enough time to make all badges & banners actionable in time for next release.
There's another week to go in this Sprint right?  Wouldn't all badges be done by then?  If not, I can remove this in the next PR or separately next week. wdyt?

**Comment by ovitrif** [comment]
> There's another week to go in this Sprint right? Wouldn't all badges be done by then? If not, I can remove this in the next PR or separately next week. wdyt?

Badges should be ready soon considering [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/16911) adds all of the currently planned ones.

I was a bit unsure if we'll get enough time to add the `onClick` behavior to all of them :)

Either way, I'm merging this PR in light of your remark here :shipit: :

> I need to extract styles, so will handle that in the next PR

**Review by ovitrif — APPROVED** [review]
Ready to merge 🚀

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/jetpackbadge/JetpackPoweredBottomSheetFragment.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    dismiss()
+                }
+            }.exhaustive
+        }
+    }
+
+    private fun showFullScreenBottomSheet(bottomSheet: FrameLayout) {
+        val layoutParams = bottomSheet.layoutParams
+        layoutParams.height = Resources.getSystem().displayMetrics.heightPixels
+        bottomSheet.layoutParams = layoutParams
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/jetpackbadge/JetpackPoweredBottomSheetFragment.kt:102]: Could this have been just:

```suggestion
bottomSheet.layoutParams.height = Resources.getSystem().displayMetrics.heightPixels
```

or am I missing something?

**Pull request merged**